core/src/main/java/com/scalar/db/storage/jdbc/JdbcAdmin.java

```diff
@@ -824,4 +848,87 @@ private String enclose(String name) {
   private String encloseFullTableName(String schema, String table) {
     return rdbEngine.encloseFullTableName(schema, table);
   }
+
+  @Override
+  public Set<String> getNamespaceNames() throws ExecutionException {
+    try (Connection connection = dataSource.getConnection()) {
+      String selectQuery =
+          "SELECT * FROM " + encloseFullTableName(metadataSchema, NAMESPACES_TABLE);
+      Set<String> namespaces = new HashSet<>();
+      try (Statement statement = connection.createStatement();
+          ResultSet resultSet = statement.executeQuery(selectQuery)) {
+        while (resultSet.next()) {
+          namespaces.add(resultSet.getString(NAMESPACE_COL_NAMESPACE_NAME));
+        }
+        return namespaces;
+      }
+    } catch (SQLException e) {
+      // An exception will be thrown if the namespace table does not exist when executing the select
+      // query
+      if (rdbEngine.isUndefinedTableError(e)) {
+        return Collections.emptySet();
+      }
+      throw new ExecutionException("Getting the namespace names failed", e);
+    }
+  }
+
+  private void createNamespacesTableIfNotExists(Connection connection) throws ExecutionException {
+    try {
+      createMetadataSchemaIfNotExists(connection);
+      String createTableStatement =
+          "CREATE TABLE "
+              + encloseFullTableName(metadataSchema, NAMESPACES_TABLE)
+              + "("
+              + enclose(NAMESPACE_COL_NAMESPACE_NAME)
+              + " "
+              + getTextType(128)
+              + ", "
+              + "PRIMARY KEY ("
+              + enclose(NAMESPACE_COL_NAMESPACE_NAME)
+              + "))";
+      createTableIfNotExists(connection, createTableStatement);
+    } catch (SQLException e) {
+      throw new ExecutionException("Creating the namespace table failed", e);
+    }
+  }
+
+  private void insertIntoNamespacesTable(Connection connection, String namespaceName)
+      throws SQLException {
+    String insertStatement =
+        "INSERT INTO "
+            + encloseFullTableName(metadataSchema, NAMESPACES_TABLE)
+            + " VALUES ('"
+            + namespaceName
```

Comment:
How about using prepared statement? Strictly speaking, this code **itself** seems to allow SQL injection. So either using prepared statement or some more SQL escape is needed, ideally.

Reply:
Thank you, I switch to prepared statements for DML statements. 
I revised it https://github.com/scalar-labs/scalardb/pull/989#discussion_r1286624632